great_expectations/core/serializer.py

```diff
@@ -0,0 +1,62 @@
+"""Serializer class interface definition.
+
+Serializers determine how to write an object to disk, json, etc.
+A serializer comprises the object destination and name e.g. YAMLReadyDictDatasourceConfigSerializer.
+A base implementation (DictConfigSerializer) is provided if no modification needs to be included for the specific object / destination pair.
+
+Typical usage example:
+
+datasource_config = DatasourceConfig(...)
+serializer = DictConfigSerializer(schema=datasourceConfigSchema)
+serialized_value = serializer.serialize(datasource_config)
+"""
+
+import abc
+from typing import Optional, Union
+
+from great_expectations.core.configuration import AbstractConfig
+from great_expectations.marshmallow__shade import Schema
+
+
+class AbstractConfigSerializer(abc.ABC):
+    """Serializer interface.
+
+    Note: When mypy coverage is enhanced further, this Abstract class can be replaced with a Protocol.
+    """
+
+    def __init__(self, schema: Optional[Schema] = None) -> None:
+        """
+        Args:
+            schema: marshmallow schema defining raw serialized version of object. Optional
+                because some subclasses override with a default value.
+        """
+        self._schema = schema
```

Comment:
non-blocking suggestion.

Can we just make this attribute public? It feels like it's part of the API since subclasses are counting on it being there.